yang/frr-ospfd.yang

```diff
@@ -0,0 +1,903 @@
+      list dlist { 
+        key "name protocol"; 
+	leaf name { 
+	  type string; 
+	  description
+	    "Filter networks in routing updates."; 
+	} 
+	leaf protocol { 
+	  type frr-route-types:frr-route-types-v4; 
+	  description
+	    "Protocol."; 
+	} 
+	leaf out { 
+	  type empty; 
+	  description
+	    "In direction."; 
+	} 
+	leaf in { 
+	  type empty; 
+	  description
+	    "Out direction."; 
+	} 
+      } 
+    } 
+  }
+
+  grouping max-metric-group {
+    container max-metric { 
+      container router-lsa { 
+	  description
+	    "Advertise own Router-LSA with infinite distance (stub router)."; 
+        leaf administrative { 
+	  type empty; 
+	  description
+	    "Administratively applied,  for an indefinite period."; 
+	} 
+	leaf on-shutdown { 
+	  type uint8 { 
+	    range "5..100"; 
+	  } 
+	  description
+	    "Advertise stub-router prior to full shutdown of OSPF."; 
+	} 
+	leaf on-startup { 
+	  type uint32 { 
+	    range "5..86400"; 
+	  } 
+	  description
+	    "Automatically advertise stub Router-LSA on startup of OSPF."; 
+	} 
+      } 
+    } 
+  }
+
+  grouping mpls-te-group {
+    container mpls-te { 
+      leaf on { 
+        type empty; 
+	description
+	  "Enable the MPLS-TE functionality."; 
+
+      } 
+      leaf router-address { 
+        type inet:ipv4-address; 
+	description
+	  "Stable IP address of the advertising router."; 
+      } 
+      container inrer-as { 
+        leaf as { 
+	  type empty; 
+	  description
+	    "AS native mode self originate INTER_AS LSA with Type 11 (as flooding scope)."; 
+	} 
+	leaf area { 
+	  type inet:ipv4-address; 
+	  description
+	    "AREA native mode self originate INTER_AS LSA with Type 10 (area flooding scope)."; 
+	} 
+      } 
+    } 
+  }
+
+  grouping ospf-group { 
+    container ospf { 
+      leaf abr-type { 
+        type enumeration { 
+	  enum "cisco" { 
+	    value 1; 
+	    description
+	      "Alternative ABR,  cisco implementation.";
+	  } 
+	  enum "ibm" { 
+	    value 2; 
+	    description
+	      "Alternative ABR,  IBM implementation.";
+	  } 
+	  enum "shortcut" { 
+	    value 3; 
+	    description
+	      "Shortcut ABR.";
+	  } 
+	  enum "standard" { 
+	    value 4; 
+	    description
+	      "Standard behavior (RFC2328).";
+	  } 
+	} 
+      } 
+      leaf opaque-lsa { 
+        type empty; 
+	description
+	  "Enable the Opaque-LSA capability (rfc2370)";
+      } 
+      leaf rfc1583compatibility { 
+        type empty; 
+	description
+	  "Enable the RFC1583Compatibility flag.";
+      } 
+      leaf router-id { 
+        type inet:ipv4-address; 
+	description
+	  "Router-id for the OSPF process.";
+      } 
+      leaf write-multiplier { 
+        type uint8 { 
+	  range "1..100"; 
+	} 
+	description
+	  "Write multiplier.";
+      } 
+    } 
+  }
+
+  grouping timer-group {
+    container timers { 
+      leaf refresh-interval { 
+        type uint16 { 
+          range "10..1800"; 
+        } 
+	description
+	  "Timer value in seconds.";
```

Comment:
what timer? :)

Reply:
That's a poor description indeed.

Also, please use `units seconds;`.